Docker/StorageApi/Writer.php

```diff
@@ -551,18 +601,58 @@ public function tagFiles(array $configuration)
         }
     }
 
+    /**
+     * @param array $tableInfo
+     * @param array $config
+     */
     public function validateAgainstTable($tableInfo = [], $config = [])
     {
         // primary key
-        if (count($config["primary_key"]) > 0 || count($tableInfo["primaryKey"]) > 0) {
-            if (count(array_diff($tableInfo["primaryKey"], $config["primary_key"])) > 0 ||
-                count(array_diff($config["primary_key"], $tableInfo["primaryKey"])) > 0
+        $configPK = self::normalizePrimaryKey($config["primary_key"]);
+        if (count($configPK) > 0 || count($tableInfo["primaryKey"]) > 0) {
+            if (count(array_diff($tableInfo["primaryKey"], $configPK)) > 0 ||
+                count(array_diff($configPK, $tableInfo["primaryKey"])) > 0
             ) {
-                $pkMapping = join(", ", $config["primary_key"]);
+                $pkMapping = join(", ", $configPK);
                 $pkTable = join(", ", $tableInfo["primaryKey"]);
                 $message = "Output mapping does not match destination table: primary key '{$pkMapping}' does not match '{$pkTable}' in '{$config["destination"]}'.";
                 throw new UserException($message);
             }
         }
     }
+
+    /**
+     * @param array $pKey
+     * @return array
+     */
+    public static function normalizePrimaryKey(array $pKey)
+    {
+        return array_unique(array_filter($pKey, function ($col) {
+            if ($col != '') {
+                return true;
+            }
+            return false;
+        }));
+    }
+
+    /**
+     * @param array $features
+     * @param array $tableInfo
+     * @param array $config
+     * @return bool
+     */
+    public static function modifyPrimaryKeyDecider(array $features, array $tableInfo, array $config)
+    {
+        if (!in_array("docker-runner-output-mapping-adaptive-pk", $features)) {
+            return false;
+        }
+        $configPK = self::normalizePrimaryKey($config["primary_key"]);
+        if (count($tableInfo["primaryKey"]) != count($configPK)) {
```

Comment:
no, do týhle podmínky to vstupuje s tím, že ty pole maj stejnou délku, tj např `["Id", "Name"]` a ["FirstName", "LastName"]. pokud ty dvě pole nemaj průnik o stejný dýlce jako jako jakýkoliv z těch polí (a obě jsou stejně dlouhý), tak ty pole nejsou stejný, right?